### PR TITLE
Restore MUMPS_ARM_STR to configure_petsc.sh for manual Apple Silicon builds and elif clang-min definition changes

### LINKED ISSUE #25991: Create a min-clang container
## Reason
We want to support our Minimum Clang target using an Apptainer.

## Design
I am thinking a `ALTERNATE_FROM = clang-min` modification to mpich.def.

## Impact
Version control the Minimum Clang def file, as we do with Modern Clang.

### PR DESCRIPTION
**Pull request by cticenhour**
Removing this has been causing failures in CI. Also, increment conda packages.

Refs #25287

This PR also brings in the content from #25992:

>Build our minimum version of supported Clang
>
>Closes https://github.com/idaholab/moose/issues/25991

### CODE REVIEW

**Review by milljm — APPROVED** [review]
sigh (that this is needed)... 👍

**Comment by cticenhour** [comment]
Python failure in Apple Silicon testing is related to the civet client test, and therefore unrelated to this change. Tagging @loganharbour for another set of eyes on the package version updates.

**Comment by loganharbour** [comment]
Can you guys work on combining with https://github.com/idaholab/moose/pull/25992?

**Review by loganharbour — CHANGES_REQUESTED** [review]
FYI still not passing

File: conda/libmesh/conda_build_config.yaml
Comment (cticenhour) [inline conda/libmesh/conda_build_config.yaml:5]:
```suggestion
  - moose-libmesh-vtk 9.2.6 build_7
```

**Review by loganharbour — APPROVED** [review]
Reminder to remove the `ONLY_ON_PR_NUM` options in the recipes for these, and also to add steps to the "Apptainer transfer" recipe on master to also transfer these images.

**Review by loganharbour — CHANGES_REQUESTED** [review]
Peacock was bumped which will require a versioner_hashes change due to a moose-dev bump.

**Comment by milljm** [comment]
I need to solve the issue with channel transfer before I can allow this onto next.

**Comment by milljm** [comment]
>Reminder to remove the ONLY_ON_PR_NUM options in the recipes for these, and also to add steps to the "Apptainer transfer" recipe on master to also transfer these images.

will do, as soon as its merged...

**Comment by milljm** [comment]
@brandonlangley may have some idea to improve our `moose.def` file. I figure that may as well wind up in this perhaps. Or, if we are chomping at the bits to get this in, the `moose.def` changes can wait for another PR...

**Comment by milljm** [comment]
Okay, scratch that... we will remove the WASP section in another PR, and not hold this one up. Being that it is redundant, means we can be a little lazy and do it later...

**Review by loganharbour — APPROVED** [review]

**Pull request merged**
